packages/strapi-plugin-upload/controllers/Upload.js

```diff
@@ -15,31 +17,80 @@ module.exports = {
    */
 
   index: async (ctx) => {
+    console.log(ctx.request.body.files);
+    const config = await strapi.store({
+      environment: strapi.config.environment,
+      type: 'plugin',
+      name: 'upload'
+    }).get({key: 'provider'});
+
+    if (!config.enabled) {
+      return ctx.badRequest(null, ctx.request.admin ? [{ messages: [{ id: 'Upload.status.disabled' }] }] : 'Upload is disabled!');
+    }
+
     const Service = strapi.plugins['upload'].services.upload;
 
-    const files = await Service.buffurize(ctx.request.body.files);
+    // transform stream files to buffer
+    const files = await Service.bufferize(ctx.request.body.files);
 
-    await Service.upload(files);
+    for (var i = 0; i < files.length; i++) {
+      if (files[i].size > config.sizeLimit) {
+        return ctx.badRequest(null, ctx.request.admin ? [{ messages: [{ id: 'Upload.status.sizeLimit' }] }] : 'One of file is bigger than limit size!');
```

Comment:
Maybe the error could be more precise and say what's the file which is too large?